### DISCUSSION TITLE
rest.operationpage and rest.tagpage

### DISCUSSION THREAD

**Discussion by cjlotz** [dotnet/docfx#9570] [Q&A]
Hello

In the latest version of docfx, what is the status of the `rest.operationpage` and `rest.tagpage` packages?  I see these specific packages been deprecated, but I've noticed new [Docfx.Build.OperationLevelRestApi](https://www.nuget.org/packages/Docfx.Build.OperationLevelRestApi) and [Docfx.Build.TagLevelRestApi](https://www.nuget.org/packages/Docfx.Build.TagLevelRestApi) packages which seem to be replacements for these deprecated packages?  I can however find no reference on how to use them?  I'm specifically interested to understand how to use them with the `docfx build` command

**Comment by cjlotz** [comment]
From my testing it seems like the `Docfx.Build.TagLevelRestApi.dll` assembly needs to be located in the same folder as where the docfx tool is installed?  If it is in the same folder, it seems to be automatically loaded and used.  However, as we are using docfx as a dotnet tool, the directory is something like `C:\Users\CarelL\.nuget\packages\docfx\2.74.1\tools\net7.0\any`.  Is there not a way to bootstrap the `dotnet docfx build` command to pass it a directory from which to load these extensions?  @yufeih  any suggestions?  I couldn't find anything by looking at source code/existing docs.

**Reply by yufeih** [reply]
It'll also load from [`<custom-template-dir>/plugins` directory](https://github.com/dotnet/docfx/blob/main/src/Docfx.App/Helpers/DocumentBuilderWrapper.cs#L37)

**Reply by cjlotz** [reply]
Thanks @yufeih .  I got it working using the custom template `plugins` directory but it seems that all depedencies for the `Docfx.Build.TagLevelRestApi.dll` needs to be in the same directory as well?  If I don't add these dependencies in same directory, I get the following error showing that the assembly is not loaded due to requested types being missing?

![image](https://github.com/dotnet/docfx/assets/1300847/6571123c-2fb1-441e-a26a-1327b425aff9)

Is this to be expected?  I was hoping that these would be resolved from the restored dotnet tools directory, i.e. `C:\Users\CarelL\.nuget\packages\docfx\2.74.1\tools\net7.0\any` in my instance.  

Here's a screen shot of all the dependencies I had to add for the plugin in my custom `plugins` directory 

![image](https://github.com/dotnet/docfx/assets/1300847/812e24e9-0e12-4c28-a3bf-f43329a90904)

**Reply by yufeih** [reply]
>I was hoping that these would be resolved from the restored dotnet tools directory,

That was my expectation as well, something must be broken.

**Reply by filzrev** [reply]
> If I don't add these dependencies in same directory, I get the following error showing that the assembly is not loaded due to requested types being missing?

On my environment. 
Plugin assemblies can be loaded successfully.

**Test Steps**
1. Copy 2-plugin dlls to `samples\seed\template\plugins` directory. (I'm tested with `net60` version DLLs)
2. Run `docfx build --verbose` command on `samples\seed` directory.
3. Following logs are displayed.

> Loading plug-ins and post-processors...
> Searching custom plugins in directory C:\Projects\GitHub\filzrev\docfx\samples\seed\template\plugins...
> Scanning assembly file C:\Projects\GitHub\filzrev\docfx\samples\seed\template\plugins\Docfx.Build.OperationLevelRestApi.dll...
> Scanning assembly file C:\Projects\GitHub\filzrev\docfx\samples\seed\template\plugins\Docfx.Build.TagLevelRestApi.dll...

**Workaround**
I don't know why some assemblies are not loaded before plugin DLL loading on reported environment.
Thought it can be resolved by adding `Known DLLs` loading steps. (like [older version of docfx plugin loadings](https://github.com/dotnet/docfx/blob/v2.64.0/src/Microsoft.DocAsCode.App/Helpers/DocumentBuilderWrapper.cs#L53-L62))
